### PR TITLE
Fixes MAUI tests and updates pipeline to use VS2022 Preview

### PR DESCRIPTION
**Pull request by Sweekriti91**
- [x] Fixes Broken .NET MAUI Extension tests
- [x] Adds documentation for .NET MAUI extension implementation
- [x] Fixes .NET MAUI extension upgrade steps to work with rc-2 .NET MAUI release
- [x] Updates AzDo Pipeline to use VS2022 preview internal pool VMs
- [x] confirm tests pass.

### CODE REVIEW

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/Templates/MauiAndroid/MainApplication.cs [hunk 1]
```diff
@@ -2,15 +2,18 @@
...
 	{
 		public MainApplication(IntPtr handle, JniHandleOwnership ownership)
 			: base(handle, ownership)
 		{
 		}
-	}
-}
\ No newline at end of file
+
+        protected override MauiApp CreateMauiApp() => MauiProgram.CreateMauiApp();
```
Comment (brandonh-msft) [inline src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/Templates/MauiAndroid/MainApplication.cs:17]: It seems like we have to always add this boilerplate; I couldn't dive in to figure out why this doesn't just exist by default in `MauiApplication` but feels like it should.

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiUtilties.cs [hunk 2]
```diff
@@ -14,14 +14,18 @@ internal static class MauiUtilties
...
+                return MauiProjectType.Maui;
             }
             else if (components.HasFlag(ProjectComponents.MauiAndroid))
             {
                 return MauiProjectType.MauiAndroid;
             }
+            else if (components.HasFlag(ProjectComponents.MauiiOS))
+            {
+                return MauiProjectType.MauiiOS;
+            }
```
Comment (brandonh-msft) [inline src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiUtilties.cs:28]: Trying to grok the intent of this change and what we want here. i.e. What happens if `components` has `Android` and `Maui` but not `iOS`?

Feels like what we're after might be
```csharp
if (!components.HasFlag(ProjectComponents.Maui))
{
    if (components.HasFlag(ProjectComponents.MauiiOS))
    {
        return MauiProjectType.MauiiOS;
    }
    else if (components.HasFlag(ProjectComponents.MauiAndroid))
    {
        return MauiProjectType.MauiAndroid;
    }
}
```

right?

File: README.md [hunk 3]
```diff
@@ -74,6 +74,8 @@ The following tutorials will give you a sense of how to upgrade ASP.NET, Windows
 - [Upgrade a Windows Forms App to .NET 6 with the .NET Upgrade Assistant](https://docs.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-winforms-framework)
 - [Upgrade a WPF App to .NET 6 with the .NET Upgrade Assistant](https://docs.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-wpf-framework)
 
+To upgrade Xamarin.Forms to .NET MAUI using upgrade-assistant, please follow steps [here](docs/maui_support.md).
```
Comment (maddymontaquila) [inline README.md:77]: Maybe "Learn more about upcoming support for Xamarin.Forms to .NET MAUI upgrades"? Don't want to make it too final sounding 🤣

File: docs/maui_support.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# Migrating from Xamarin.Forms to .NET MAUI
+
+upgrade-assistant include preview support for Xamarin.Forms to .NET MAUI migration. Please note this tool only has preview support and more features coming soon. This tool helps with the migration progress but you will still require to make manual changes to the solution to make it run successfully with .NET MAUI.
```
Comment (maddymontaquila) [inline docs/maui_support.md:3]: "We are currently building out support for upgrade-assistant to handle Xamarin.Forms to .NET MAUI migrations. Please note...."?

File: docs/maui_support.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+## Supported Project types and minimum requirements
+
+- Xamarin.Forms app using Xamarin.Forms 5.0 or higher
+  - ✅ iOS project
+  - ✅ Android project
+  - ❌ UWP project 
+  - ❌ iOS App extension and other extension types for example Share Extension project
```
Comment (maddymontaquila) [inline docs/maui_support.md:11]: maybe say ex. watchOS, tvOS?

File: docs/maui_support.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+- Complete environment setup for .NET MAUI including all necessary workloads installed. Confirm that your environment can build and run a blank .NET MAUI template app. Be sure to follow all steps mentioned in the [Getting Started](https://github.com/dotnet/maui/wiki#getting-started). 
+- Use version control, create a backup of current code before starting the upgrade process
```
Comment (maddymontaquila) [inline docs/maui_support.md:14]: "We recommend using version control to create a backup of your current codebase before starting the upgrade process"?

File: docs/maui_support.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+Currently supported steps:
+
+- csproj updates
+    - update csproj files to follow net6.0 format and add/remove relevant project properties.
+    - add relevant `TargetFrameworks` to csproj file
+
+- nuget updates
+    - Removes known packages not supported by .NET MAUI
+    - Adds known ports of libraries for .NET MAUI
```
Comment (maddymontaquila) [inline docs/maui_support.md:26]: Maybe bullets on which ones we know for sure? It's just Skia, MAUI graphics, XF, Xam essentials, right? MAUI community toolkit?

File: docs/maui_support.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+After running the upgrade-assistant tool, you still have to make manual changes to your projects. Complete the rest of the migration process manually, by following: 
+
+- [Manual Migration Guide](https://github.com/dotnet/maui/wiki/Migrating-from-Xamarin.Forms-(Preview))
+- [.NET MAUI Documentation](https://docs.microsoft.com/en-us/dotnet/maui/)
+- [Samples](https://github.com/dotnet/maui/wiki/Migrating-from-Xamarin.Forms-(Preview)#samples)
+
+### Issues
+
+For .NET MAUI specific issues, or issues/feedback around migrating Xamarin.Forms apps to .NET MAUI, lease File Issues [here](https://github.com/maddymontaquila/maui-migration-samples/issues/new?assignees=&labels=&template=trial-migration-template.md&title=[MIGRATION]+Your+migration+name+here).
```
Comment (maddymontaquila) [inline docs/maui_support.md:42]: *please

Comment (maddymontaquila) [inline docs/maui_support.md:42]: Maybe "If you attempt an upgrade, please file an issue [here] to let us know how it went and what issues you ran into!"

**Comment by sunandabalu** [comment]
@Sweekriti91 Thanks for taking care of this. Overall the change LGTM, I don't have anything more to add other than the comments already on your PR.

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiUtilties.cs (hunk 2 above)
Comment (Sweekriti91) [reply src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiUtilties.cs:28]: As part of the migration from [Xamarin.Forms](https://docs.microsoft.com/en-us/xamarin/get-started/what-is-xamarin-forms) to [.NET MAUI](https://docs.microsoft.com/en-us/dotnet/maui/what-is-maui), which is what we are adding this extension to upgrade-assistant to help in the process, we are allowing current [Xamarin.Forms](https://docs.microsoft.com/en-us/xamarin/get-started/what-is-xamarin-forms) users to keep the project structure as is and migrate to [.NET MAUI](https://docs.microsoft.com/en-us/dotnet/maui/what-is-maui). There are two types of project types in .NET MAUI :

- Single Project: Single project where we have only one project in the solution with multiple TFMs and each platform implementation is in a folder within this project.

- Multiple Project Heads : This is similar to the project structure in Xamarin.Forms application which have 3 projects in the solution, a main Xamarin.Forms project and a platform specific iOS and Android Project which platform specific code. 

Currently for Xamarin.Forms projects the recommended guidance for migration is to maintain the project heads and migrate to .NET MAUI while maintaining this project structure. The guide can be found [here](https://github.com/dotnet/maui/wiki/Migrating-from-Xamarin.Forms-(Preview)) which explains in complete detial how to migrate while maintaining this project structure and not going directly to a single project format. 

Coming back to your comment, therefore there is no inclusive MAUI component type. Depending on either if it is the MAUI shared project, or each platform specific project, there are different changes to the project properties, different template files to be added. So, I don't believe the change you suggested can work here. Please let me know if there are any questions to this?  I am also including links to 2 sample projects that are also mentioned in the guide that show the final product of migration while including the project heads :

https://github.com/Sweekriti91/ArtAuction/tree/maui-projecthead

https://github.com/maddymontaquila/PlantLady/tree/maui-migrate

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiUtilties.cs [hunk 9]
```diff
@@ -14,14 +14,18 @@ internal static class MauiUtilties
         public static async Task<MauiProjectType> IsMauiProject(IProject project, CancellationToken token)
```
[the remark this answers is not in the document]
Comment (Sweekriti91) [reply src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiUtilties.cs:14]: I agree with this, `GetMauiProjectTypeForProject` is a better name and I will make this change and mark this resolved after I do it. Thanks!

[the remark this answers is not in the document]
Comment (Sweekriti91) [reply src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiUtilties.cs]: Updated name!

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/Templates/MauiAndroid/MainApplication.cs (hunk 1 above)
Comment (Sweekriti91) [reply src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/Templates/MauiAndroid/MainApplication.cs:17]: this is a template file we add to a .NET MAUI Android project. The template file comes from the .NET MAUI default template. I believe this comment is more towards the .NET MAUI implementation and I would suggest adding this issue in the [.NET MAUI Github](https://github.com/dotnet/maui). The purpose of this extension is to help with the migration process from Xamarin.Forms project to .NET MAUI project and this template file is newly required in a .NET MAUI project and as mentioned before, it comes from the default .NET MAUI template. 

More information about .NET MAUI can also be found in the documentation [here](https://docs.microsoft.com/en-us/dotnet/maui/). Details about Migration and steps for migration can be found [here](https://github.com/dotnet/maui/wiki/Migrating-from-Xamarin.Forms-(Preview)).

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiUtilties.cs (hunk 2 above)
Comment (brandonh-msft) [reply src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiUtilties.cs:28]: OK so what I think I'm hearing you say is that there either exists a single project - Android or iOS - or all 3; no mix of 2. Is that correct?

Comment (Sweekriti91) [reply src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiUtilties.cs:28]: Yes that is correct, there is no mix of 2 at the moment 😄

**Comment by Sweekriti91** [comment]
@brandonh-msft  @sunandabalu  if there are no other open comments for this PR, could one of you approve it please? 😄  Thank you in advance!

**Review by sunandabalu — APPROVED** [review]

**Pull request merged**